### PR TITLE
websocket basics implemented

### LINKED ISSUE #5: Websocket Routes
Since a lot of heavy lifting will be done by the Python backend, we should have native and easy to use websocket support

Starlette already has support for websockets, so it should just be a matter of creating a child class for them that we can modify as necessary
https://www.starlette.io/websockets/

Tasks:
- [x] Create a `QuickHTMLWSRoute` class
- [x] Make it a child class of Starlette's `WebSocketRoute`

### PR DESCRIPTION
**Pull request by moseye**
roast me

closes #5

### CODE REVIEW

**Review by ThomasJRyan — DISMISSED** [review]

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]
Am I crazy or did the unit tests disappear?

**Review by ThomasJRyan — APPROVED** [review]

**Comment by moseye** [comment]
> Am I crazy or did the unit tests disappear?

must've been the wind

File: tests/test_ws.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
+from starlette.testclient import TestClient
+from starlette.websockets import WebSocket  # Added import
+from quickhtml.routing import QuickHTMLWSEndpoint
+from quickhtml import QuickHTML
+
+
+def test_ws():
+    test_app = QuickHTML()
```
Comment (rashidnhm) [inline tests/test_ws.py:8]: Not needed within this PR, but we probably want this to become a fixture in the near future to make things easier

**Review by rashidnhm — APPROVED** [review]
Looks good! Haven't worked with WS in a while so a little rusty on that.

Would it be helpful to have docs to show how this hotreload works? Doesn't have to be part of this PR

File: src/quickhtml/routing.py [hunk 2]
```diff
@@ -112,4 +114,83 @@ def add_route(
...
+    ) -> None:
+        """
+        Add multiple routes to the routing table.
+
+        Args:
+            routes (list[tuple[str, typing.Callable[[Request], typing.Awaitable[Response] | Response]]]):
+                A list of tuples containing the URL path pattern and the endpoint function.
+
+        Returns:
+            None: This method does not return anything.
```
[the remark this answers is not in the document]
Comment (moseye) [reply src/quickhtml/routing.py:136]: Nah not needed - I can remove it. Was left over from the generated docs

**Review by ThomasJRyan — APPROVED** [review]

**Pull request merged**